src/tiled/createtemplatetool.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+ * createtemplatetool.cpp
+ * Copyright 2017, Thorbjørn Lindeijer <thorbjorn@lindeijer.nl>
+ * Copyright 2017, Mohamed Thabet <thabetx@gmail.com>
+ *
+ * This file is part of Tiled.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "createtemplatetool.h"
+
+#include "mapdocument.h"
+#include "mapobject.h"
+#include "mapobjectitem.h"
+#include "maprenderer.h"
+#include "objecttemplate.h"
+#include "snaphelper.h"
+#include "tile.h"
+#include "utils.h"
+
+using namespace Tiled;
+using namespace Tiled::Internal;
+
+CreateTemplateTool::CreateTemplateTool(QObject *parent)
+    : CreateObjectTool(parent)
+{
+    QIcon icon(QLatin1String(":images/24x24/insert-image.png"));
+    icon.addFile(QLatin1String(":images/48x48/insert-image.png"));
+    setIcon(icon);
+    Utils::setThemeIcon(this, "insert-image");
+    languageChanged();
+}
+
+void CreateTemplateTool::mouseMovedWhileCreatingObject(const QPointF &pos, Qt::KeyboardModifiers modifiers)
+{
+    const MapRenderer *renderer = mapDocument()->renderer();
+
+    const QSize imgSize = mNewMapObjectItem->mapObject()->cell().tile()->size();
```

Comment:
I think it will be fine for now to not bother with the centering at all. On the upside, it will give a clear indication to the user as to where the object is actually positioned, which will be beneficial also when this anchor point eventually becomes configurable.